src/opustrainer/trainer.py

```diff
@@ -484,10 +501,47 @@ def _load_modifier(self, modifier_entry: Dict[str, Any]) -> Modifier:
         except KeyError:
             raise CurriculumLoaderError(f"unknown modifier '{name}'")
         try:
-            return modifier(**settings)
+            args = self._match_modifier_signature(modifier, settings, basepath)
+            return modifier(**args)
         except Exception as exc:
             raise CurriculumLoaderError(f"could not initialize modifier '{name}': {exc!s}") from exc
 
+    def _match_modifier_signature(self, fn:Type[Modifier], settings:Dict[str, Any], basepath:str) -> Dict[str,Any]:
```

Comment:
wow, it's like you are writing C++ here.

Reply:
I don't think I could do this in C++, not enough reflection APIs. But if I could do this with templates, I'd probably would have had better type guarantees than whatever my python LSP is coming up with.